File: POO/poo11048216/Aulas/aula15/Conta/Conta.cpp
#ifndef CONTA_H
#include "Conta.h"
#define CONTA_H
#endif

Conta::Conta()
{   }

Conta::Conta(std::string cpf, std::string nome)
{
    this->setNumero(Conta::geraNumero(10));
    this->setCpf(cpf);
    this->setTitular(nome);
    this->setSaldo(0);
    this->setBloqueada(false);
}

void Conta::setNumero(std::string num)
{
    this->numero = num;
}

void Conta::setTitular(std::string t)
{
    this->titular = t;
}

void Conta::setCpf(std::string cpf)
{
    this->cpf = cpf;
}

void Conta::setBloqueada(bool b)
{
    this->bloqueada = b;
}

void Conta::setSaldo(float s)
{
    this->saldo = s;
}

std::string Conta::getNumero()
{
    return this->numero;
}

std::string Conta::getTitular()
{
    return this->titular;
}

std::string Conta::getCpf()
{
    return this->cpf;
}

bool Conta::isBloqueada()
{
    return this->bloqueada;
}

float Conta::getSaldo()
{
    return this->saldo;
}

bool Conta::saque(float valor)
{
    if(this->getSaldo() >= valor && !this->isBloqueada())
    {
        this->setSaldo(this->getSaldo()-valor);
        return true;
    }
    return false;
}

bool Conta::deposito(float valor)
{
    if(!this->isBloqueada())
    {
        this->setSaldo(this->getSaldo()+valor);
        return true;
    }
    return false;
}

void Conta::extrato()
{
    std::cout << "============== EXTRATO ============== " << std::endl;
    std::cout << "NUMERO ..: " << this->getNumero() << std::endl;
    std::cout << "CPF .....: " << this->getCpf() << std::endl;
    std::cout << "TITULAR .: " << this->getTitular() << std::endl;
    std::cout << "SALDO ...: " << this->getSaldo() << std::endl;
    std::cout << "===================================== " << std::endl;
}

std::string Conta::geraNumero(int n)
{
    std::string num;
    int i;
    srand(time(NULL));
    for(i=0; i<n; i++)
    {
        num += (rand()%10)+'0';
    }
    return num;
}

File: POO/poo11048216/Aulas/aula16.2/JConta/Conta.java
abstract class Conta
{
    private String numero;
    private String cpf;
    private String titular;
    private boolean bloqueada;
    private float saldo;
    
    public Conta()
    {   }
    
    public Conta(String cpf, String nome)
    {
        this.setNumero(Conta.geraNumero(10));
        this.setTitular(nome);
        this.setCpf(cpf);
        this.setSaldo(0);
        this.setBloqueada(false);
    }
    
    public void setNumero(String n)
    {
        this.numero = n;
    }
    
    public void setTitular(String t)
    {
        this.titular = t;
    }
    
    public void setCpf(String c)
    {
        this.cpf = c;
    }
    
    public void setBloqueada(boolean b)
    {
        this.bloqueada = b;
    }
    
    public void setSaldo(float s)
    {
        this.saldo = s;
    }
    
    public String getNumero()
    {
        return this.numero;
    }
    
    public String getTitular()
    {
        return this.titular;
    }
    
    public String getCpf()
    {
        return this.cpf;
    }
    
    public boolean isBloqueada()
    {
        return this.bloqueada;
    }
    
    public float getSaldo()
    {
        return this.saldo;
    }
    
    public boolean saque(float v)
    {
        if(this.getSaldo() >= v && !this.isBloqueada())
        {
            this.setSaldo(this.getSaldo()-v);
            return true;
        }
        return false;
    }
    
    public boolean deposito(float v)
    {
        if(!this.isBloqueada())
        {
            this.setSaldo(this.getSaldo()+v);
            return true;
        }
        return false;
    }
    
    public void extrato()
    {
        System.out.println("============= EXTRATO =============");
        System.out.println("NUMERO ..: " + this.getNumero());
        System.out.println("CPF .....: " + this.getCpf());
        System.out.println("TITULAR .: " + this.getTitular());
        System.out.println("SALDO ...: " + this.getSaldo());
        System.out.println("===================================");
    }
    
    public abstract void correcao();
    
    public static String geraNumero(int n)
    {
        String num="";
        int i;
        java.util.Random rd = new java.util.Random();
        for(i=0; i<n; i++)
        {
            num += (char)((rd.nextInt(10))+48);
        }
        return num;
    }
}